modules/Core/src/main/java/org/terasology/core/debug/AnimationScreen.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.core.debug;
+
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.logic.characters.CharacterMovementComponent;
+import org.terasology.logic.characters.StandComponent;
+import org.terasology.logic.characters.WalkComponent;
+import org.terasology.logic.common.InspectionToolComponent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.math.geom.Quat4f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.assets.skeletalmesh.SkeletalMesh;
+import org.terasology.rendering.logic.SkeletalMeshComponent;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.widgets.*;
+
+import java.util.ArrayList;
+
+/**
+ * A Tool for testing the animations of entities.
+ * Animation speed can be edited with this utility
+ */
+public class AnimationScreen extends CoreScreenLayer {
+
+    private static final Logger logger = LoggerFactory.getLogger(AnimationScreen.class);
+    @In
+    private EntityRef entityRef;
```

Comment:
This here doesn't make sense - `EntityRef` isn't any sort of Singleton service, so it can't be injected. That's for `System` type classes that are shared for use by other things.  You should just make it a regular variable in the method where it gets used